src/main/main.cpp

```diff
@@ -33,6 +33,8 @@ int main(int argc, char *argv[]) {
 			cxxopts::value<string>())
 		("d,debug", "Enable the debugger",
 			cxxopts::value<bool>()->default_value("false"))
+        ("i,record", "Record Control Inputs",
+            cxxopts::value<bool>()->default_value("false"))
```

Comment:
Indentation is off here, toggle visible whitespace and make it consistent with the other ones.

Reply:
I think we should also have a related parameter with the file path to which to record.

Reply:
I don't know what's the issue but my IDE(Clion) does not play well with the indentation. The file has perfect indentation in VSCode but while doing git diff, I can see the mismatched indentation.

Reply:
Are you sure you've tried enabled "show whitespace" feature? You can see the indentation error when you highlight the text with the cursor on GitHub. The new lines have been indented with spaces, but the ones above are indented with tabs.

Your editor seems to be set by default to insert spaces instead of tabs.

Reply:
Yeah, I switched to tabs and it worked like a charm. Phew, that was honestly satisfying!